### PR TITLE
test(poolmanager): expand ChargeTakerFee tests

### PR DESCRIPTION
**Pull request by DongLieu**
Closes: #6633

## What is the purpose of the change

> Add a description of the overall background and high level changes that this PR introduces

*(E.g.: This pull request improves documentation of area A by adding ....*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/poolmanager/taker_fee_test.go
Comment (mattverse) [inline x/poolmanager/taker_fee_test.go:37]: Would it be possible to make this a simple boolean field and then move the 

```
				s.App.BankKeeper.SendCoins(s.Ctx, s.TestAccs[nonWhitelistedSenderIndex], s.TestAccs[whitelistedSenderIndex], coins)

```

logic inside the actual test logic?

**Review by mattverse — COMMENTED** [review]
Left seme questions, mostly lgtm!
(Thanks for catching & adding other cases besides send coins :) )

File: x/poolmanager/taker_fee_test.go
Comment (mattverse) [reply x/poolmanager/taker_fee_test.go:75]: @DongLieu

**Review by mattverse — APPROVED** [review]
LGTM

**Comment by DongLieu** [comment]
The test changes in [PR6804](https://github.com/osmosis-labs/osmosis/pull/6804/files#diff-a27409c7fb75ff7750bd8cbf302aa0d21341f1ce5981904180dae4705dcb990aR111-R129) after merging, so you should review this PR. @mattverse

**Pull request merged**
